forge/ee/lib/billing/index.js

```diff
@@ -70,6 +82,50 @@ module.exports.init = async function (app) {
                 cancel_url: `${app.config.base_url}/team/${team.slug}/applications`
             }
 
+            // Need to ensure the subscription contains all of the expected items
+            // to correlate with the teamType. This includes:
+            // - Team Plan item - already added above
+            // - Device item
+            // - An item for each instance type
+
+            // Check if a device item is required
+            const deviceBillingIds = await teamType.getDeviceBillingIds()
+            if (deviceBillingIds.product) {
```

Comment:
I'm hesitant this this is near-duplicating the logic of `updateTeamDeviceCount` (and below) `updateTeamInstanceCount`. I'd much prefer we have one single place like a BillingService that handles all of this kind of logic, but appreciate that release is tomorrow so don't want to block on refactoring...

Reply:
Given that `updateTeamInstanceCount` and `updateTeamInstanceCount` both by outputting line items needed for the subscription, I can see a reasonable route where we have a single helper class that calculates the appropriate line items for a subscription, then we update/push them as required.

https://github.com/FlowFuse/flowfuse/blob/0500dc63c7d745dbfc508d84a76dea2cdb1fb83b/forge/ee/lib/billing/index.js#L269

https://github.com/FlowFuse/flowfuse/blob/0500dc63c7d745dbfc508d84a76dea2cdb1fb83b/forge/ee/lib/billing/index.js#L170

Reply:
This can be a follow up though, so I propose we open an issue and unblock this PR.

Reply:
Agreed - the logic is close but not quite identical and didn't want to fall into the hole of trying to pull out the common parts.